kyo-core/shared/src/main/scala/kyo/Meter.scala

```diff
@@ -239,8 +262,23 @@ object Meter:
         protected def dispatch[A, S](v: => A < S): A < (S & IO)
         protected def onClose(): Unit
 
+        private inline def withReentry[A, S](inline reenter: => A < S)(acquire: AllowUnsafe ?=> A < S): A < (IO & S) =
+            if reentrant then
+                acquiredMeters.use { meters =>
+                    if meters.contains(this) then reenter
+                    else IO.Unsafe(acquire)
+                }
+            else
+                IO.Unsafe(acquire)
+
+        private inline def withAcquiredMeter[A, S](inline v: => A < S) =
+            if reentrant then
+                acquiredMeters.update(_ + this)(v)
```

Comment:
Couldn't this be inlined to line 269?

Reply:
This method is used at the end when a computation is being effectively dispatched to update the local with the acquired meter. `withReentry` is used at another point at the start to check if the fiber is reentering. Between these two points, the computation might stay suspended in case there are no available permits